scripts/west_commands/fetchers/__init__.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+import logging
+import os
+from pathlib import Path
+
+from fetchers.core import ZephyrBlobFetcher
+
+_logger = logging.getLogger('fetchers')
+
+def _import_fetcher_module(fetcher_name):
+    try:
+        importlib.import_module(f'fetchers.{fetcher_name}')
+    except ImportError as ie:
+        # Fetchers are supposed to gracefully handle failures when they
+        # import anything outside of stdlib, but they sometimes do
+        # not. Catch ImportError to handle this.
+        _logger.warning(f'The module for fetcher "{fetcher_name}" '
+                        f'could not be imported ({ie}). This most likely '
+                        'means it is not handling its dependencies properly. '
+                        'Please report this to the zephyr developers.')
+
+# We import these here to ensure the BlobFetcher subclasses are
+# defined; otherwise, BlobFetcher.get_fetchers() won't work.
+
+# Those do not contain subclasses of ZephyrBlobFetcher
+name_blocklist = ['__init__', 'core']
+
+fetchers_dir = Path(__file__).parent.resolve()
+for f in [f for f in os.listdir(fetchers_dir)]:
```

Comment:
This "suffers" from the same limitation as the runner discovery method - it only supports in-tree fetchers. While I do admit that out-of-tree fetchers are likely a more curious thing than out-of-tree runners, it would be nice if we could derive a better pattern for this.

Reply:
It does, yes. But in this case it’s not an issue because an out of tree fetcher just doesn’t really make sense. You might as well commit your binaries to you OoT module.